projects/netgrif-components-core/src/lib/task/services/finish-task.service.ts

```diff
@@ -23,6 +23,7 @@ import {AfterAction} from '../../utility/call-chain/after-action';
 import {ChangedFieldsService} from '../../changed-fields/services/changed-fields.service';
 import {EventService} from '../../event/services/event.service';
 import {ChangedFieldsMap} from '../../event/services/interfaces/changed-fields-map';
+import {TaskEventOutcome} from '../../event/model/event-outcomes/task-outcomes/task-event-outcome';
```

Comment:
The 5 changed task services seem to have a similar structure and implementation. It might be worth investigating a refactor (in a separate issue) where the common functionality would be extracted into an abstract class of some sort.